### ISSUE TITLE
Zuul#add-host-header=true not work,  ProxyRequestHelper ignore "host" header

### ISSUE THREAD

**Issue by huge0612** [spring-cloud/spring-cloud-netflix#3171]
my Zuul configurations:
```
zuul:
  sensitive-headers:
  add-host-header: true
```
`add-host-header` property should make Zuul send proxy request with header `host`,but i found these in ProxyRequestHelper：
```
	public boolean isIncludedHeader(String headerName) {
		String name = headerName.toLowerCase();
		RequestContext ctx = RequestContext.getCurrentContext();
		if (ctx.containsKey(IGNORED_HEADERS)) {
			Object object = ctx.get(IGNORED_HEADERS);
			if (object instanceof Collection && ((Collection<?>) object).contains(name)) {
				return false;
			}
		}
		switch (name) {
		case "host":
		case "connection":
		case "content-length":
		case "content-encoding":
		case "server":
		case "transfer-encoding":
		case "x-application-context":
			return false;
		default:
			return true;
		}
	}
```
and this:
```
public MultiValueMap<String, String> buildZuulRequestHeaders(
			HttpServletRequest request) {
		RequestContext context = RequestContext.getCurrentContext();
		MultiValueMap<String, String> headers = new HttpHeaders();
                // ...
		Map<String, String> zuulRequestHeaders = context.getZuulRequestHeaders();
		for (String header : zuulRequestHeaders.keySet()) {
                        // zuulRequestHeaders constains 'host' header, but isIncludedHeader() method ignore it
			if (isIncludedHeader(header)){
				headers.set(header, zuulRequestHeaders.get(header));
			}
		}
                // ...
		return headers;
	}
```
but ,in Spring Cloud Finchley.Realse version, like this:
```
public MultiValueMap<String, String> buildZuulRequestHeaders(
			HttpServletRequest request) {
		RequestContext context = RequestContext.getCurrentContext();
		MultiValueMap<String, String> headers = new HttpHeaders();
                // ...
		Map<String, String> zuulRequestHeaders = context.getZuulRequestHeaders();
		for (String header : zuulRequestHeaders.keySet()) {
                                 // add all zuulRequestHeaders , include 'host' header
				headers.set(header, zuulRequestHeaders.get(header));
		}
                // ...
		return headers;
	}
```

**Comment by ryanjbaxter** [comment]
So this worked in Edgware.SR4?  

Can you provide a sample that reproduces it?

**Comment by huge0612** [comment]
sorry for reply late, i miss your message. 
I write a demo project. [link](https://github.com/huge0612/demo)

**Comment by ryanjbaxter** [comment]
Fixed via 6e8ef02
